test/hotspot/jtreg/gc/testlibrary/Helpers.java

```diff
@@ -101,7 +102,7 @@ public static void compileClass(String className, Path root, String source) thro
                 .addToolArg(sourceFile.toAbsolutePath().toString());
 
         ProcessBuilder pb = new ProcessBuilder(jar.getCommand());
-        OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        OutputAnalyzer output = ProcessTools.executeProcess(pb);
```

Comment:
The pattern
```java
OutputAnalyzer output = ProcessTools.executeProcess(jcmd.getCommand());
```
is used above, but an explicit `ProcessBuilder` is used here. Maybe one style be picked used?

`test/hotspot/jtreg/gc/arguments/TestUseCompressedOopsFlagsWithUlimit.java` does something similar to this as well.

Reply:
I've changed this to skip using an explicit ProcessBuilder. However, that hit a problem that some ProcessTools.executeProcess overloads are declared to throw Throwable instead of Exception. Instead of fixing that inside this PR, or creating a workaround for this, I created a separate, trivial PR to update those functions. Please see #17240. I've merged the branch for that PR into this PR so that we can test the updated changes.